tempesta_fw/http.c

```diff
@@ -3367,24 +3368,39 @@ tfw_http_req_process(TfwConn *conn, TfwStream *stream, const TfwFsmData *data)
 	parser = &stream->parser;
 	actor = TFW_MSG_H2(req) ? tfw_h2_parse_req : tfw_http_parse_req;
 
-	r = ss_skb_process(skb, off, trail, actor, req, &req->chunk_cnt,
+	r = ss_skb_process(skb, off, data_left, actor, req, &req->chunk_cnt,
 			   &parsed);
 	req->msg.len += parsed;
-	curr_skb_trail = off + parsed + trail < skb->len ? 0 : trail;
+	data_left -= parsed;
 	TFW_ADD_STAT_BH(parsed, clnt.rx_bytes);
 
 	T_DBG2("Request parsed: len=%u next=%pK parsed=%d msg_len=%lu"
 	       " ver=%d res=%d\n",
 		 skb->len, skb->next, parsed, req->msg.len, req->version, r);
 
+	switch (r) {
+	case TFW_POSTPONE:
```

Comment:
Why do we chop the message on receiving *each* chunk? I believe we should chop it only once.

Besides the logic issue, the construction is quite ugly - why not to use `if (r == TFW_POTPONE || r == TFW_PASS)`?

Reply:
> Why do we chop the message on receiving each chunk?

Legacy. :smiley: 
As you mentioned earlier, we should probably completely remove the need to chop skbs here by preparing them earlier. But at the moment there are offsets, and we address them this way.

> why not to use if

Changed to `if`. I was trying to mimic the switch-case below for easier comparison.

Reply:
There were changes that removed the need to handle heading and trailing parts here, so this discussion is no longer relevant.